runtime/lua/vim/lsp/semantic_tokens.lua

```diff
@@ -0,0 +1,158 @@
+local M = {}
+local util = require "vim.lsp.util"
+local ns = vim.api.nvim_create_namespace("lsp-semantic-tokens")
+
+local semantic_tokens = {}
+
+local function modifiers_from_number(x, modifiers_table)
+  local function get_bit(n, k)
+    -- Based on/from https://stackoverflow.com/a/26230537
+    -- (n >> k) & 1
+    return _G.bit.band(_G.bit.rshift(n, k), 1)
+  end
+
+  local modifiers = {}
+  for i = 0, #modifiers_table - 1 do
+    local bit = get_bit(x, i)
+    if bit == 1 then
+      table.insert(modifiers, 1, modifiers_table[i + 1])
+    end
+  end
+
+  return modifiers
+end
+
+M.token_map = {
+  type = "LspType",
+  class = "LspClass",
+  enum = "LspEnum",
+  interface = "LspInterface",
+  struct = "LspStruct",
+  typeParameter = "LspTypeParameter",
+  parameter = "LspParameter",
+  variable = "LspVariable",
+  property = "LspProperty",
+  enumMember = "LspEnumMember",
+  event = "LspEvent",
+  ["function"] = "LspFunction",
+  method = "LspMethod",
+  macro = "LspMacro",
+  keyword = "LspKeyword",
+  modifier = "LspModifier",
+  comment = "LspComment",
+  string = "LspString",
+  number = "LspNumber",
+  regexp = "LspRegexp",
+  operator = "LspOperator"
+}
+
+M.modifiers_map = {
+  declaration = "LspDeclaration",
+  definition = "LspDefinition",
+  readonly = "LspReadonly",
+  static = "LspStatic",
+  deprecated = "LspDeprecated",
+  abstract = "LspAbstract",
+  async = "LspAsync",
+  modification = "LspModification",
+  documentation = "LspDocumentation",
+  defaultLibrary = "LspDefaultLibrary"
+}
```

Comment:
TODO: find strategy that's both conventional and flexible. Should align to already available strategy for tree-sitter

Reply:
Is somehow aligned with tree-sitter. Open Question: how handle a combination of token and modifiers.`modifiers_cache[ft][token.type..m] or  modifiers_cache[ft][m]`?

Reply:
is the "Lsp" prefix really needed? Or should we try to map to builtin highlight/syntax groups (`:help highlight-groups
`)? (And potentially introduce _new_ builtin highlight/syntax groups...)

Reply:
I am strongly in favor of defining new, basic highlight groups for these new kinds of tokens; these could (and should) be linked by default to the classical vim groups to make them visible out of the box. That allows a decent OOTB experience while still allowing maximal customization (which I hold to be a major yardstick for Neovim).

I'm not going to bikeshed on the actual names, although they seem fine to me.

(*Ceterum censeo* the builtin standard groups are starting to show their age and are becoming increasingly restrictive for Neovim's rapidly expanding use. Might be worth revisiting at some point, especially in connection with tree-sitter graduating to first-class syntax highlighter -- whenever that will be.)

Reply:
I've made the experience that color scheme authors love to have control over the tree-sitter highlight groups while other color scheme still work perfect with the default mapping.

I think that creating a good default mapping would still take some discussions so it might be a good task for a follow-up PR.